pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -542,3 +557,59 @@ func implicitWaypointPolicies(ctx krt.HandlerContext, Waypoints krt.Collection[W
 		return ptr.Of(w.Namespace + "/" + policy)
 	})
 }
+
+func gatewayUID(gw model.NetworkGateway) string {
+	return "NetworkGateway/" + string(gw.Network) + "/" + gw.Addr + "/" + strconv.Itoa(int(gw.HBONEPort))
+}
+
+// convertGateway always converts a NetworkGateway into a Workload
+func convertGateway(gw model.NetworkGateway) model.WorkloadInfo {
+	wl := &workloadapi.Workload{
+		Uid:            gatewayUID(gw),
+		ServiceAccount: gw.ServiceAccount.Name,
+		Namespace:      gw.ServiceAccount.Namespace,
+		Network:        gw.Network.String(),
+	}
+
+	if ip, err := netip.ParseAddr(gw.Addr); err == nil {
+		wl.Addresses = append(wl.Addresses, ip.AsSlice())
+	} else {
+		wl.Hostname = gw.Addr
+	}
+
+	return model.WorkloadInfo{Workload: wl}
+}
+
+func (a *index) getNetworkGateway(id string) []model.NetworkGateway {
```

Comment:
I think this isn't marked dependent for any collection but the NetworkGatewayWorkloads.. we will want to do that here to cover all the types of workloads

Reply:
Sorry not sure I fully understand. There is one trigger for "depends on network". Its already called on all the workload types.

Is the recommendation to refactor it so we call MarkDependant in one place (getNetworkGateway)? Or is there a place its missing

Reply:
Oh, yeah disregard. I guess it might be good to put that MarkDependant here if calling that twice doesn't have side-effects.